### ISSUE TITLE
CustomKenel/VersionIssue

### ISSUE THREAD

**Issue by dduquesne** [rancher/os#2438]
**RancherOS Version: 1.1.4**

**baremetal**

Hello,

I tried to build the kernel using the dedicated process.
Then i have build RancherOS by cloning the 1.1.4 branch on github.
It works but when i boot i have the version information below.
What could be wrong.

![image](https://user-images.githubusercontent.com/12520088/43929104-c9af5248-9c33-11e8-83c4-d69783571c07.png)

**Comment by niusmallnan** [comment]
How did you make it?

It seems like that you changed the `KERNEL_VERSION_amd64=4.9.78-rancher` in `Dockerfile.dapper`.

**Comment by dduquesne** [comment]
I followed the kernel build manual one the Rancher kernel project. I changed the kernel version into the dapper file, i customized the kernel config file and i applied a patch using the patch folder. I put the result into the assset folder of the OS project before building it. It works but i am worried about 2 things :
- the rancher suffix on the kernel version. When i try to install perf , it try to find the package linux-tool-<>

**Comment by dduquesne** [comment]
I tried to change the dapper file to remove this rancher suffix on the kernel version.
I just change :

ARG OS_REPO=rancher
ARG KERNEL_TAG
ARG KERNEL_VERSION=${KERNEL_TAG}-${OS_REPO}

to

ARG OS_REPO=rancher
ARG KERNEL_TAG
ARG KERNEL_VERSION=${KERNEL_TAG}

It does not work compilation fail.
What is the aim of this suffix ?

**Comment by dduquesne** [comment]
Do you think that it can help to solve my issue or it sound like a not important detail ? (I am unable to install linuxtool package)

**Comment by niusmallnan** [comment]
If you want to use perf, you should compile it from source.
The perf is a kernel tool, and RancherOS has its own kernel.

If I have time I will try to compile it.

**Comment by kingsd041** [comment]
@dduquesne I have a temporary solution, you can try to install perf in the following way, I can install successfully in my environment
```
# ros -v
version v1.4.1 from os image rancher/os:v1.4.1
# ros console switch ubuntu
# apt update && apt install gcc make flex bison -y
# curl -L -o linux-`uname -r`-src.tgz  https://github.com/rancher/os-kernel/releases/download/v`uname -r`/linux-`uname -r`-src.tgz
# tar -xz -f linux-`uname -r`-src.tgz
# cd v`uname -r`/tools/perf
# make && make install 

# ./perf version
perf version 4.14.67
```

**Comment by kingsd041** [comment]
Close it.
If you still have doubts, please reopen it and let me know.

**Comment by dduquesne** [comment]
Hello @kingsd041 

Sorry for the delay.
Your answer came several weeks after and i have not seen it till today.
I had gave up after a while, a little bit desperate and i have given it another try today as the need for this tool is still there.
I have tried your "temporary solution" and i have perf running on my custom rancheros  :+1: 
Many thanks for your help :1st_place_medal: 
It will be very useful to move forward on my current project.
